models/user/roleSchema.js

```diff
@@ -0,0 +1,36 @@
+import Joi from 'joi';
+import { Schema } from 'mongoose';
+import { roles } from '../../constants/index.js';
+import { idValidationSchema } from '../../helpers/validation.js';
+
+export const roleSchema = new Schema(
+  {
+    name: {
+      type: String,
+      enum: Object.values(roles),
+      required: true,
+    },
+    id: {
+      type: Schema.Types.ObjectId,
+      default: null,
+    },
+  },
+  { _id: false }
+);
+
+export const roleValidationSchema = Joi.object({
+  name: Joi.string()
+    .valid(...Object.values(roles))
+    .not(roles.USER)
+    .required(),
+  id: Joi.alternatives().try(Joi.allow(null), idValidationSchema).required(),
+});
+
+export const getDefaultRoles = () => {
+  const { USER, ...otherRoles } = roles; // eslint-disable-line no-unused-vars
```

Comment:
`lodash.omit(roles, 'USER')` - will create a new object without USER property
https://lodash.com/docs/4.17.15#omit

Reply:
Okay, I'll use this library. Thanx

Reply:
Fixed this issue. I made the default `roles: [{"user" : "userId"}]` in order to return only one object  req.roleIds from the auth middleware instead of two objects, as it was in the previous version of the middleware (`req.userId = `, `req.roles =`)

right now getDefaultRoles has the next view
`export const getDefaultRoles = function () { return [ { name: roles.USER, id: this._id, }, ]; };`